Add GitCode Provider (#1020)
aspnet-contrib/AspNet.Security.OAuth.Providers: docs/gitcode.md

Reviewer: I'm not sure we need this, as it just makes a comment but doesn't otherwise show anything different in terms of code/options.

Author: Perhaps this will offer some help to developers who directly stumble upon this and are unfamiliar with its usage. At the very least, they can copy and paste it for immediate use. Therefore, I’m inclined to keep this documentation.

# Integrating the GitCode Provider

## Example

```csharp
services.AddAuthentication(options => /* Auth configuration */)
        .AddGitCode(options =>
        {
            options.ClientId = "my-client-id";
            options.ClientSecret = "my-client-secret";

>             // It appears that GitCode does not support setting the scopes through query strings.
>             // The configuration of scopes (App permissions) is instead managed
>             // within the OAuth app management on the official web.

Reviewer: ```suggestion
            // GitCode does not support setting scopes via requests to the API.
            // The configuration of scopes (App permissions) are instead managed
            // within the OAuth app management in GitCode itself.
```
